src/util/invariant.cpp

```diff
@@ -129,11 +115,20 @@ void check_invariant(
 #else
     out << "Backtraces not supported\n" << std::flush;
 #endif
+}
 
-
-#ifdef CPROVER_INVARIANT_PRINT_STACK_TRACE
-    abort();
-#else
-    throw invariant_failedt(out.str());
-#endif
+/// Returns a backtrace
+///
+/// \param file : C string giving the name of the file.
+/// \param function : C string giving the name of the function.
+/// \param line : The line number of the invariant
+/// \return backtrace with a file / function / line header.
+std::string get_backtrace(
+  const char * const file,
+  const char * const function,
```

Comment:
I am aware that this implies further changes; but given the effort by @reuk to get rid of plain pointers we should stick with that direction.